p2p/p2p.go

```diff
@@ -408,6 +408,11 @@ func (s *Service) persistPeers() error {
 
 	store := s.host.Peerstore()
 	peers := store.Peers()
+
+	peers = utils.Filter(peers, func(peerID peer.ID) bool {
+		return peerID != s.host.ID()
+	})
+
```

Comment:
Nice catch!

Perhaps it's better to do just this
```suggestion
	peers := utils.Filter(store.Peers(), func(peerID peer.ID) bool {
		return peerID != s.host.ID()
	})
```